frontend/lib/Views/AdminPage/groups_view.dart

```diff
@@ -164,22 +172,95 @@ class _GroupsViewState extends State<GroupsView> {
               Container(
                 alignment: Alignment.centerRight,
                 child: Icon(
-                  Icons.arrow_right_alt,
+                  Icons.arrow_forward_ios_rounded,
                   color: color,
+                  size: 25,
                 ),
               ),
             ],
           ),
         ),
       );
 
-  void _onHoverButton(GroupCard groupCard, String text, Color color) {
-    setState(() {
-      if (text == 'Members') {
-        groupCard.membersButtonColor = color;
+  Container _buildDeleteContainer(GroupCard groupCard) => Container(
+        padding: const EdgeInsets.symmetric(vertical: 20, horizontal: 20),
+        child: OutlinedButton(
+          style: ButtonStyle(
+            side: MaterialStateProperty.all(
+              const BorderSide(width: 1.5, color: Colors.red),
+            ),
+            padding: MaterialStateProperty.all(
+              const EdgeInsets.all(20),
+            ),
+            foregroundColor: MaterialStateProperty.resolveWith<Color>(
+              (Set<MaterialState> states) =>
+                  states.contains(MaterialState.hovered)
+                      ? Colors.white
+                      : Colors.red,
+            ),
+            backgroundColor: MaterialStateProperty.resolveWith<Color>(
+              (Set<MaterialState> states) =>
+                  states.contains(MaterialState.hovered)
+                      ? Colors.red
+                      : Colors.white,
+            ),
+            alignment: Alignment.centerLeft,
+          ),
+          child: const Icon(Icons.delete_outline_outlined, size: 30),
+          onPressed: () => _onDeletePressed(groupCard),
+        ),
+      );
+
+  FloatingActionButton _buildAddButton() => FloatingActionButton(
+        onPressed: () => _onCreatePressed(groupListProvider),
+        backgroundColor: floatingButtonColor,
+        child: const Icon(Icons.add),
+      );
+
+  void _onCreatePressed(groupListProvider) {
+    showCreateGroupModal(context, createGroup);
+  }
+
+  void _onDeletePressed(GroupCard groupCard) {
+    showAlertDialog(
+      context,
+      'Delete ' + groupCard.name,
+      "You are about to delete group with all of its' saved permissions.",
+      () => deleteGroup(groupCard.id),
+    );
+  }
+
+  void deleteGroup(int id) async {
+    await widget.gateway
+        .deleteGroup(id)
+        .then(
+          (value) => {
+            if (value)
+              {
+                groupListProvider.delete(id),
+              }
+          },
+        )
+        .catchError((error) {
+      buildSnackbar('Cannot delete group', context);
+    });
+  }
+
+  void createGroup(String name) async {
+    await widget.gateway.createGroup(name).then((value) {
+      if (value.id != -1) {
```

Comment:
`-1` w tym kontekście to trochę magic number. Może zrobić stałą `EMPTY_GROUP_ID`, lub coś podobnego?